From 9879d1c2b78caafc757b883b0fe8ccc8e7c4548f Mon Sep 17 00:00:00 2001
From: Maksim Ivanov <emaxx@google.com>
Date: Sun, 29 May 2022 16:24:39 +0200
Subject: [PATCH] Make sure AraKiriHotPlug false on libusb startup (#135)

Reset the AraKiriHotPlug flag to false when starting libusb hotplug
search.

This is not needed on the initial process startup, because the flag is
initialized to false, but allows to implement unit tests that need to
restart the previously stopped threads within the memory of the same
process.
---
 src/hotplug_libusb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/hotplug_libusb.c b/src/hotplug_libusb.c
index 7462c863..3c50f526 100644
--- a/src/hotplug_libusb.c
+++ b/src/hotplug_libusb.c
@@ -550,6 +550,7 @@ LONG HPSearchHotPluggables(void)
 {
 	int i;
 
+	AraKiriHotPlug = FALSE;
 	for (i=0; i<PCSCLITE_MAX_READERS_CONTEXTS; i++)
 	{
 		readerTracker[i].status = READER_ABSENT;
